src/shared/hooks/useDataApi.ts

```diff
@@ -67,8 +75,8 @@ const reducer = (state: State, action: Action): State => {
   }
 };
 
-const useDataApi = (url?: string): State => {
-  const [state, dispatch] = useReducer(reducer, { loading: !!url });
+function useDataApi<T>(url?: string | null): State<T> {
```

Comment:
I tried to avoid switching from arrow function to this, but then the code would look stranger.
TypeScript didn't let me write a nice arrow function with generics. See https://stackoverflow.com/a/45576880